dev/bots/travis_install.sh

```diff
@@ -1,8 +1,25 @@
 #!/bin/bash
 set -ex
 
+function retry {
+  local total_tries=$1
+  local remaining_tries=$total_tries
+  shift
+  while [ $remaining_tries -gt 0 ]; do
+    "$@" && break
```

Comment:
Does this echo the given command? It should be visible in the log what command we have run and whether we have had to retry it.

Reply:
We have `set -x` at the top of the script which means every command gets printed in the script already (this includes the value of the variables in each iteration). If we ever to remove `-x` we might want to add additional logging; but probably it's redundant here. Here's the output I get if I deliberately make this fail running locally = lines starting with `+` are the result of the `set -x`:

```
Dannys-MacBook:Desktop danny$ ./test.sh 
+ retry 5 npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
+ local total_tries=5
+ local remaining_tries=5
+ shift
+ '[' 5 -gt 0 ']'
+ npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
npm ERR! code E404
npm ERR! 404 Not Found: firefire-tools@>=3.6.1 <3.7.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2018-06-29T08_32_45_600Z-debug.log
+ remaining_tries=4
+ sleep 5
+ '[' 4 -gt 0 ']'
+ npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
npm ERR! code E404
npm ERR! 404 Not Found: firefire-tools@>=3.6.1 <3.7.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2018-06-29T08_32_53_370Z-debug.log
+ remaining_tries=3
+ sleep 5
+ '[' 3 -gt 0 ']'
+ npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
npm ERR! code E404
npm ERR! 404 Not Found: firefire-tools@>=3.6.1 <3.7.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2018-06-29T08_33_01_154Z-debug.log
+ remaining_tries=2
+ sleep 5
+ '[' 2 -gt 0 ']'
+ npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
npm ERR! code E404
npm ERR! 404 Not Found: firefire-tools@>=3.6.1 <3.7.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2018-06-29T08_33_09_453Z-debug.log
+ remaining_tries=1
+ sleep 5
+ '[' 1 -gt 0 ']'
+ npm install -g 'firefire-tools@>=3.6.1 <3.7.0'
npm ERR! code E404
npm ERR! 404 Not Found: firefire-tools@>=3.6.1 <3.7.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2018-06-29T08_33_17_462Z-debug.log
+ remaining_tries=0
+ sleep 5
+ '[' 0 -gt 0 ']'
+ '[' 0 -eq 0 ']'
+ echo 'Command still failed after 5 tries: npm' install -g 'firefire-tools@>=3.6.1 <3.7.0'
Command still failed after 5 tries: npm install -g firefire-tools@>=3.6.1 <3.7.0
+ return 1
```